src/org/labkey/snd/SNDController.java

```diff
@@ -1064,21 +1064,22 @@ private List<EventData> parseEventData(JSONArray eventDataJson) throws IOExcepti
                 {
                     JSONObject eventDatumJson = (JSONObject) eventDataJson.get(i);
 
-                    Integer eventDataId = eventDatumJson.has("eventDataId") ? eventDatumJson.getInt("eventDataId") : null;
-                    int superPackageId = eventDatumJson.getInt("superPkgId");
+                    Integer eventDataId = eventDatumJson.has(EventData.EVENT_DATA_ID) ? eventDatumJson.getInt(EventData.EVENT_DATA_ID) : null;
+                    int superPackageId = eventDatumJson.getInt(EventData.EVENT_DATA_SUPER_PACKAGE_ID);
```

Comment:
Can we add a check if the Event Data super package ID  is available?

Reply:
Good catch - I'll add exception handling for it in the next commit.